src/dotnet/commands/dotnet-migrate/ProjectJsonParser.cs

```diff
@@ -0,0 +1,81 @@
+﻿using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.Cli
+{
+    /// <summary>
+    /// Parses select data from a project.json without relying on ProjectModel.
+    /// Used to parse simple information.
+    /// </summary>
+    internal class ProjectJsonParser
+    {
+        public static string SdkPackageName => "Microsoft.DotNet.Core.Sdk";
+
+        public string SdkPackageVersion { get; }
+
+        public ProjectJsonParser(JObject projectJson)
+        {
+            SdkPackageVersion = GetSdkPackageVersion(projectJson);
+        }
+
+        private string GetSdkPackageVersion(JObject projectJson)
+        {
+            var sdkPackageNode = SelectJsonNodes(projectJson, property => property.Name == SdkPackageName).First();
```

Comment:
>SdkPackageName [](start = 91, length = 14)

It seems like all of the SelectJsonNode code could be simplified into a single Linq statement:

```
            return projectJson
                .Descendants()
                .OfType<JProperty>()
                .FirstOrDefault(property => property.Name == SdkPackageName)
```